src/ZODB/scripts/repozo.py

```diff
@@ -73,6 +73,12 @@
         Note:  for the stdout case, the index file will **not** be restored
         automatically.
 
+    -w
+    --with-verification
+        Verify on the fly the backup files on recovering. This option runs
+        the same checks as when repozo is run in -V/--verify mode, and
+        allows to verify and recover a backup in one single step.
```

Comment:
I wonder if it's worth mentioning that if verification fails, the output file is left in a partial, inconsistent state.

Reply:
That's an interesting issue you are mentioning. From a user experience point of view, do you think that deleting the partially restored zodb would be a good idea ? Otherwise user may use it (ie: after a disaster recovery) without noticinig data is missing. More especially, this partially restored file is valid according to the zodb format.

Reply:
I was torn about it, and therefore didn't suggest it.  The partial file may be useful for trying to investigate/fix the corruption (although I'm not sure how).  But the risk of the user accidentally using a truncated Data.fs is real.

Perhaps it would be best add a `.tmp` suffix to the output file, and rename it to the final name only if everything's fine.

Reply:
I like the while-restoring-extension idea. Just to throw in a few more suffix ideas: `.part` ? `.incomplete` ? I am not sure how strong the dot-three extension format rule is these days.